src/ModelContextProtocol/TokenProgress.cs

```diff
@@ -29,3 +29,27 @@ public void Report(ProgressNotificationValue value)
         }, CancellationToken.None);
     }
 }
+
+//TODO: Create an intermediary interface to handle common operations (cancel, ping, and progress).
+//This will allow a single class to handle this implementation.
+internal sealed class ClientTokenProgress(IMcpClient client, ProgressToken progressToken) : IProgress<ProgressNotificationValue>
```

Comment:
Let's put these in their own files. And if we're going to call this one ClientTokenProgress, the other should be called ServerTokenProgress.